Web.md

```diff
@@ -20,6 +20,44 @@ WebAssembly's [modules](Modules.md) allow for natural [integration with
 the ES6 module system](Modules.md#integration-with-es6-modules) and allow
 synchronous calling to and from JavaScript.
 
+### Function Names
+
+A WebAssembly module imports and exports functions. WebAssembly names functions
+using arbitrary-length byte sequences. Any 8-bit values are permitted in a
+WebAssembly name, including the null byte and byte sequences that don't
+correspond to any Unicode code point regardless of encoding. The most natural
+Web representation of a mapping of function names to functions is a JS object
+in which each function is a property. Property names in JS are UTF-16 encoded
+strings. A WebAssembly module may fail validation on the Web if it imports or
+exports functions whose names do not transcode cleanly to UTF-16 according to
+the following conversion algorithm, assuming that the WebAssembly name is in a
+`Uint8Array` called `array`:
+
+```
+function convertToJSString(array)
+{
+  // Perform the actual conversion.
+  var string = "";
+  for (var i = 0; i < array.length; ++i)
+    string += String.fromCharCode(array[i]);
+  var result = decodeURIComponent(escape(string));
+  
+  // Check for errors. This will throw if 'result' contains bad characters.
+  encodeURIComponent(result);
```

Comment:
decodeURIComponent should throw.  The reason why I didn’t just rely on that is that some of the public documentation documents encodeURIComponent as throwing but doesn’t say that decodeURIComponent should throw.  But having read the spec (both ES5.1 and ES6), I no longer think that the encodeURIComponent call is necessary.

I *think* that we can remove the encodeURIComponent call.  Objections?

-Filip


> On Mar 2, 2016, at 7:26 AM, Luke Wagner <notifications@github.com> wrote:
> 
> In Web.md <https://github.com/WebAssembly/design/pull/573#discussion_r54737759>:
> 
> > +strings. A WebAssembly module may fail validation on the Web if it imports or
> > +exports functions whose names do not transcode cleanly to UTF-16 according to
> > +the following conversion algorithm, assuming that the WebAssembly name is in a
> > +`Uint8Array` called `array`:
> > +
> > +```
> > +function convertToJSString(array)
> > +{
> > +  // Perform the actual conversion.
> > +  var string = "";
> > +  for (var i = 0; i < array.length; ++i)
> > +    string += String.fromCharCode(array[i]);
> > +  var result = decodeURIComponent(escape(string));
> > +  
> > +  // Check for errors. This will throw if 'result' contains bad characters.
> > +  encodeURIComponent(result);
> I should add: it has the right behavior, but I'd still like to know if the encodeURIComponent is necessary or if decodeURIComponent catches it all.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/WebAssembly/design/pull/573/files#r54737759>.
>